src/fsharp/CheckDeclarations.fs

```diff
@@ -2452,8 +2583,20 @@ let TcMutRecDefns_Phase2 (cenv: cenv) envInitial bindsm scopem mutRecNSInfo (env
                 // Does not need to be hidden behind a lang version as it needs to be possible to
                 //     implement protected interface methods in lower F# versions regardless if it's a DIM or not.
                 match tyconOpt with
-                | Some _ when declKind = DeclKind.ModuleOrMemberBinding ->
-                    MakeInnerEnvForTyconRef envForDecls tcref false
+                | Some tycon when declKind = DeclKind.ModuleOrMemberBinding ->
+                    let envForDecls = MakeInnerEnvForTyconRef envForDecls tcref false
+                    if tycon.IsUnionTycon then
+                        let shouldAugment =
```

Comment:
Please extract this `shouldAugment` logic out into a function (put it in AddAugmentationDeclarations)

Reply:
Done!